h2o-py/h2o/expr.py

```diff
@@ -73,7 +72,7 @@ class ExprNode(object):
 
     # Magical count-of-5:   (get 2 more when looking at it in debug mode)
     #  2 for _get_ast_str frame, 2 for _get_ast_str local dictionary list, 1 for parent
-    MAGIC_REF_COUNT = 5 if sys.gettrace() is None else 7  # M = debug ? 7 : 5
+    MAGIC_REF_COUNT = 4 if sys.gettrace() is None else 6  # M = debug ? 6 : 4
```

Comment:
Could we explain this a bit better? Also "Magical count-of-5:" should be maybe changed to "Magical count-of-4:"

Reply:
This MAGIC_REF_COUNT thing is pure evil enfleshed. Once upon a time I spent several days debugging problems caused by it.
Ideally there should be a *clear* and *detailed* explanation of what this does and what the "magical" counts are. I suspect current comment are out-of-sync with reality...

Reply:
agree @st-pasha , there is no well defined semantics of that field. I would like to replace that but slightly different implementation.